presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/SelectionContext.java

```diff
@@ -13,17 +13,28 @@
  */
 package com.facebook.presto.spi.resourceGroups;
 
+import java.util.Optional;
+
 import static java.util.Objects.requireNonNull;
 
 public final class SelectionContext<T>
 {
     private final ResourceGroupId resourceGroupId;
     private final T context;
+    private final Optional<ResourceGroupSelectionContext> resourceGroupSelectionContext;
+
+    public SelectionContext(ResourceGroupId resourceGroupId, T context, ResourceGroupSelectionContext resourceGroupSelectionContext)
+    {
+        this.resourceGroupId = requireNonNull(resourceGroupId, "resourceGroupId is null");
+        this.context = requireNonNull(context, "context is null");
+        this.resourceGroupSelectionContext = requireNonNull(Optional.of(resourceGroupSelectionContext), "context is null");
```

Comment:
`Optional.of()` will never return a null. You probably want to change this to `Optional.of(requireNonNull())`